aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -10,6 +11,19 @@ namespace at {
 namespace native {
 namespace {
 
+DEFINE_DISPATCH(qcat_nhwc_stub);
+DEFINE_DISPATCH(qcat_relu_nhwc_stub);
+
+bool is_cat_nhwc_fast_path(const c10::List<Tensor>& qxs, int dim) {
+  TORCH_CHECK(qxs.size() > 0);
+  bool is_fast_path = dim == 1;
+  for (const at::Tensor& qx : qxs) {
+    is_fast_path &= qx.dim() == 4;
+    is_fast_path &= qx.is_contiguous(c10::MemoryFormat::ChannelsLast);
+  }
+  return is_fast_path;
```

Comment:
Does the code correctly handle tensors with different layouts? If x is in nhwc and y is in nchw, does concat((x,y)) work?

Reply:
Yes, it is very clear from the code that the fast path is taken if *all* tensors are 4D and *all* tensors are NHWC